### PR TITLE
feat: Add error annotation when a deprecated lambda runtime is used

### PR DESCRIPTION
**Pull request by akash1810**
## What does this change?

We should never use a deprecated or non-LTS runtime for lambdas. In this change, we add an error annotation to this effect.

Note: the annotation does not prevent the stack from being synthesised, to do this, we'd want to `throw`. I'm not sure if that's desirable behaviour atm though?

## How to test

See added tests.

## How can we measure success?

We are discouraged from using deprecated or non-LTS lambda runtimes.

## Have we considered potential risks?

We have to maintain the `DEPRECATED_RUNTIMES` list.

## Checklist

- [ ] I have listed any breaking changes, along with a migration path [^1]
- [ ] I have updated the documentation as required for the described changes [^2]

[^1]: Consider whether this is something that will mean changes to projects that have already been migrated, or to the CDK CLI tool. If changes are required, consider adding a checklist here and/or linking to related PRs.
[^2]: If you are adding a new construct or pattern, has new documentation been added? If you are amending defaults or changing behaviour, are the existing docs still valid?

### CODE REVIEW

**Comment by akash1810** [comment]
Raised this as a way to test https://github.com/guardian/cdk/pull/1006 hasn't negatively impacted the release process.

File: src/constructs/lambda/lambda.ts [hunk 1]
```diff
@@ -10,6 +10,14 @@ import type { GuStack } from "../core";
 import { AppIdentity } from "../core/identity";
 import { GuParameterStoreReadPolicyStatement } from "../iam";
 
+const DEPRECATED_RUNTIMES: Runtime[] = [
+  Runtime.NODEJS_4_3,
+  Runtime.NODEJS_6_10,
+  Runtime.NODEJS_8_10,
+  Runtime.NODEJS_10_X,
+  Runtime.NODEJS_12_X,
```
Comment (jacobwinch) [inline src/constructs/lambda/lambda.ts:18]: IIUC, this [version](https://nodejs.org/en/about/releases/) becomes EOL on 2022-04-30 - do we want to deprecate it before then?

**Comment by jacobwinch** [comment]
> Note: the annotation does not prevent the stack from being synthesised, to do this, we'd want to throw. I'm not sure if that's desirable behaviour atm though?

I guess this is something that we (or the Security team) might want to consider in the future but I think it's better to start with this less disruptive approach.

**Review by jacobwinch — APPROVED** [review]

File: src/constructs/lambda/lambda.ts (hunk 1 above)
Comment (akash1810) [reply src/constructs/lambda/lambda.ts:18]: Ah, I thought it was already deprecated! I think it's fine to do so before then as:
  - We're not blocking the synth step
  - It acts as a reminder to update _before_ the deadline
  - There aren't too many lambdas defined using GuCDK today (and 0 using this version of the library)

**Pull request merged**